config/src/main/java/com/typesafe/config/impl/ConfigBeanImpl.java

```diff
@@ -105,8 +109,18 @@
                 Method setter = beanProp.getWriteMethod();
                 Type parameterType = setter.getGenericParameterTypes()[0];
                 Class<?> parameterClass = setter.getParameterTypes()[0];
-                Object unwrapped = getValue(clazz, parameterType, parameterClass, config, originalNames.get(beanProp.getName()));
-                setter.invoke(bean, unwrapped);
+                String configPropName = originalNames.get(beanProp.getName());
+                if (configPropName == null) {
+                    configPropName = beanProp.getName();
+                }
```

Comment:
previously the code would have thrown NullPointerException if beanProp.getName wasn't in originalNames it looks like; is this case (`configPropName == null`) actually possible? Do the unit tests find a way to trigger this codepath? (there is a coverage target in the sbt build I believe to check)